pkg/store/bucket.go

```diff
@@ -1597,6 +1631,18 @@ func (r *bucketIndexReader) LabelValues(name string) []string {
 	return res
 }
 
+// LabelNames returns a list of label names.
+func (r *bucketIndexReader) LabelNames(ctx context.Context) ([]string, error) {
+	res := make([]string, 0, len(r.block.lvals))
+	for ln, _ := range r.block.lvals {
+		if ctx.Err() != nil {
```

Comment:
Mixed feelings, as IMO it is early, micro optimization that only increases code complexity.

There is no much of the point in checking ctx.Err in every place. It takes more cycles to check this IF than rest of this function not being cancelled (: I guess we wanted to save that memory in this slice when request is cancelled right? I would drop it IMO.